google-cloud-storage/src/main/java/com/google/cloud/storage/StorageClass.java

```diff
@@ -111,7 +112,11 @@ public static StorageClass valueOfStrict(String constant) {
   }
 
   /** Get the StorageClass for the given String constant, and allow unrecognized values. */
+  @JsonCreator
   public static StorageClass valueOf(String constant) {
+    if (constant == null || constant.isEmpty()) {
+      return null;
+    }
```

Comment:
Why is this necessary?

If I remove it and run the following test, the test passes:
```java
  private static final class TestXmlObject2 {
    @JacksonXmlProperty(localName = "storageClass")
    private StorageClass storageClass;
  }
  @Test
  public void testParseStringValueEnum() throws IOException {
    //language=xml
    String xml = "<TestXmlObject2>\n"
        + "  <storageClass>STANDARD</storageClass>"
        + "</TestXmlObject2>";
    InputStream in = new ByteArrayInputStream(xml.getBytes(StandardCharsets.UTF_8));
    TestXmlObject2 expected = new TestXmlObject2(StorageClass.STANDARD);
    TestXmlObject2 actual =
        xmlObjectParser.parseAndClose(in, StandardCharsets.UTF_8, TestXmlObject2.class);
    assertThat(actual).isEqualTo(expected);
  }
```